commit 36f7ae2318a2c3f8404ffe2549ca46994667c5cd
Author: Maximiliano Curia <maxy@gnuservers.com.ar>
Date:   Thu Feb 9 18:30:56 2017 +0100

    Fix public domain regex
    
    Apparently it was taking forever processing this, on the strings from a
    binary file (mostly long numbers).
    
    Also add variables to the frame to ease debugging (i for position and
    license_ for the license name, as the license_re items aren't reachable
    when using gdb).

diff --git a/decopy/matchers.py b/decopy/matchers.py
index f72bd88..264c57e 100755
--- a/decopy/matchers.py
+++ b/decopy/matchers.py
@@ -790,9 +790,9 @@ LICENSES_RES = (
                    r'no copyright on that code', re.IGNORECASE),
         'public-domain', None),
     ReLicense(
-        re.compile(r'[^ ]+ was written by(?: [^ ]+){,4}, and is placed in the '
-                   r'public domain. The author hereby disclaims copyright to '
-                   r'this source code.', re.IGNORECASE),
+        re.compile(r' was written by(?: [^ ]+){,4}, and is placed in the '
+                   r'public domain\. The author hereby disclaims copyright to '
+                   r'this source code\.', re.IGNORECASE),
         'public-domain', None),
     ReLicense(
         re.compile(
@@ -1688,13 +1688,14 @@ def find_licenses(text):
     '''
     licenses = {}
 
-    for license_re in LICENSES_RES:
+    for i, license_re in enumerate(LICENSES_RES):
+        # Keep the license name in this frame for easier debugging
+        license_ = license_re.license
         if license_re.license in licenses:
             continue
         match = license_re.re.search(text)
         if not match:
             continue
-        license_ = license_re.license
         if license_re.get_detail:
             license_ = license_re.get_detail(text, match, license_)
         licenses[license_re.license] = license_
